### PR TITLE
docs(ivy): add better documentation around debugging ivy tests

### PR DESCRIPTION
**Pull request by benlesh**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

Updating docs.

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

Just adding more information about build flags and the like.

### CODE REVIEW

File: docs/BAZEL.md [hunk 1]
```diff
@@ -68,7 +68,29 @@ new as of May 2017 and not very stable yet.
...
+### Various Flags Used For Tests
+
+If you're experiencing problems with seemingly unrelated tests failing, it may be because you're not using the proper flags with your Bazel test runs in Angular.
+
+See also: [`//tools/bazel.rc`](https://github.com/angular/angular/blob/master/tools/bazel.rc) where `--define=ivy=false` is defined as default.
+
+See also: [`./src/ivy_switch.ts`](https://github.com/angular/angular/blob/master/packages/core/src/ivy_switch.ts) for more details.
+
+- `--config=debug`: build and launch in debug mode (see [debugging](#debugging) instructions below)
+- `--define=compile=<option>` Controls if ivy or legacy mode is enabled. This is down by generating the [`src/ivy_switch.ts`](https://github.com/angular/angular/blob/master/packages/core/src/ivy_switch.ts) file from [`ivy_switch_legacy.ts`](https://github.com/angular/angular/blob/master/packages/core/src/ivy_switch_legacy.ts) (default), [`ivy_switch_jit.ts`](https://github.com/angular/angular/blob/master/packages/core/src/ivy_switch_jit.ts), or [`ivy_switch_local.ts`](https://github.com/angular/angular/blob/master/packages/core/src/ivy_switch_local.ts).
```
Comment (JoostK) [inline docs/BAZEL.md]: This is do**w**n -> This is do**n**e

File: docs/BAZEL.md [hunk 2]
```diff
@@ -68,7 +68,29 @@ new as of May 2017 and not very stable yet.
...
+    - `legacy`: (default behavior) compile against legacy renderer, e.g. `--define=compile=legacy`
```
Comment (vicb) [inline docs/BAZEL.md]: "legacy renderer` -> "View Engine" ?

File: docs/BAZEL.md [hunk 3]
```diff
@@ -68,7 +68,29 @@ new as of May 2017 and not very stable yet.
...
+    - `jit`: Compile in ivy JIT mode, e.g. `--define=compile=jit`
+    - `local`: Compile in ivy AOT move, e.g. `--define=compile=local`
+- `--test_tag_filters=<tag>`: filter tests down to tags defined in the `tag` config
+of your rules in any given `BUILD.bazel`.
+    - `ivy-jit`: Only run tests that have to do with ivy JIT, e.g. `--test_tag_filters=ivy-jit`, e.g. `--test_tag_filters=ivy-local`. For this,
```
Comment (vicb) [inline docs/BAZEL.md]: Change "Only run tests that have to do with ivy JIT"  to "This flag should be set for tests that should be excuted with ivy JIT"  ?

File: docs/BAZEL.md [hunk 4]
```diff
@@ -107,6 +129,7 @@ First time setup:
         },
 ```
 
+** Setting breakpoints directly in your code files may not work in VSCode**.
```
Comment (vicb) [inline docs/BAZEL.md]: why ? / what should I do then ?
(Do you refer to the fact that bkpoints should be added to built file in `/private/...` ?)

File: docs/BAZEL.md [hunk 5]
```diff
@@ -68,7 +68,29 @@ new as of May 2017 and not very stable yet.
 You can use [ibazel] to get a "watch mode" that continuously
 keeps the outputs up-to-date as you save sources.
 
+### Various Flags Used For Tests
+
+If you're experiencing problems with seemingly unrelated tests failing, it may be because you're not using the proper flags with your Bazel test runs in Angular.
+
+See also: [`//tools/bazel.rc`](https://github.com/angular/angular/blob/master/tools/bazel.rc) where `--define=ivy=false` is defined as default.
+
+See also: [`./src/ivy_switch.ts`](https://github.com/angular/angular/blob/master/packages/core/src/ivy_switch.ts) for more details.
```
Comment (vicb) [inline docs/BAZEL.md]: Aren't comments in `./src/ivy_switch.ts` outdated ? Shouldn't that file refer to here ? (and the same for the bazel files)

File: docs/BAZEL.md (hunk 1 above)
Comment (benlesh) [reply docs/BAZEL.md]: Thanks.

File: docs/BAZEL.md (hunk 2 above)
Comment (benlesh) [reply docs/BAZEL.md]: Thanks.

File: docs/BAZEL.md (hunk 3 above)
Comment (benlesh) [reply docs/BAZEL.md]: Yup. There were other issues there too.

File: docs/BAZEL.md (hunk 4 above)
Comment (benlesh) [reply docs/BAZEL.md]: Okay, updated.

File: docs/BAZEL.md (hunk 5 above)
Comment (benlesh) [reply docs/BAZEL.md]: Yup... updating that.

**Review by vicb — APPROVED** [review]

**Pull request closed without merging**